### PR TITLE
Adresses #211: Reset form after submitting

### PR DESCRIPTION
**Pull request by drepram**
I don't know, I guess in a way this PR addresses #211? What do I put here LoL @pushkalkatara

#### Changes proposed in this pull request:

- Since the data binds itself to `this.create_team`, I decided to just reset the value of those variable after the API has been successfully called.

- Link to live demo: http://pr-238-evalai.surge.sh 

- 
Screen Shot 2019-12-13 at 00 06 17

### CODE REVIEW

**Comment by Sanji515** [comment]
@drepram After hitting `Create Team` button, team gets created but also it shows message `Team is required`. Can you please check this?

**Comment by Sanji515** [comment]
Also @drepram After resetting the form it should also focus-out of the input.

**Comment by drepram** [comment]
I've managed to figure out how to bypass the form's validity.

The logic in the `Team Name Required` label is valid, we just need to figure out how to make the valueless input becomes valid. The solution that I've come to is that we should create a temporary value, in this case, a space character (" ").

When the input is clicked, there will be a ternary operator that would reset the input to no value ("") if it's only value is a space character (" "). If the value is not a space character, the input's value will stick to that value and would not reset it.

And on my machine, the input focuses out when I submit them. I can't reproduce the issue that you've stated to me, Hence, I declare it fixed.

Feedbacks are wanted.

PS: Can anyone approve https://github.com/Cloud-CV/CloudCV/pull/116/files? It's been 3 days and no one has left a single comment on it.

**Comment by Sanji515** [comment]
> When the input is clicked, there will be a ternary operator that would reset the input to no value ("") if it's only value is a space character (" "). If the value is not a space character, the input's value will stick to that value and would not reset it.

This looks hacky, can you please think of some other solution? You can remove the current team name required validation and can add a new one when the `Create Team` function gets called from the typescript.

> And on my machine, the input focuses out when I submit them. I can't reproduce the issue that you've stated to me

Please look closely, when you submit then placeholder `team name` remains smaller on some height.
Hint: https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/app/components/publiclists/teamlist/teamlist.component.ts#L21
Please take a look at these (^).

**Comment by drepram** [comment]
I have done the following things:

- Unfocus input when form is submitted
- Add another variable that would balance the 'Team Name Required' logic

Please take a look, feedbacks are requested.

File: src/app/components/publiclists/teamlist/teamlist.component.html [hunk 1]
```diff
@@ -43,7 +43,7 @@
                      #teamName="ngModel" [(ngModel)]="create_team['team_name']" required>
               <span class="form-icon form-icon-dark"><i class="fa fa-user"></i></span>
               <label for="name" [class.active]="isnameFocused" data-error="wrong" data-success="right">Team Name*</label>
-              <div class="wrn-msg text-highlight" *ngIf="teamName.invalid && (teamName.dirty || teamName.touched || createTeamForm.submitted)">Team Name Required</div>
+              <div class="wrn-msg text-highlight" *ngIf="validateTeamName && teamName.invalid && (teamName.dirty || teamName.touched || createTeamForm.submitted)">Team Name Required</div>
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamlist.component.html]: As it is only a `required` validation then we can remove `teamName.invalid && (teamName.dirty || teamName.touched || createTeamForm.submitted` from the `if` condition and can use a single variable to toggle the `Team Name Required` message.

Make default of this variable to `false` and toggle this variable inside `createTeamSubmit` function on the `else` case when form is not valid if `team_name` is empty.

Also please change variable name to `isTeamNameRequired`

Please let me know if I'm not clear.

Comment (drepram) [reply src/app/components/publiclists/teamlist/teamlist.component.html]: I've done the changes you've requested, may I resolve this conversation?

**Review by Sanji515 — COMMENTED** [review]
You need to re-initialize `isTeamNameRequired` variable to its default state cause it will fail if user first hits `Create Team` without any input and then tries to create team.

Also can you please change the validation message to `Team name is required`?

**Comment by drepram** [comment]
I've change the validation message and changed back `isTeamNameRequired` to `false` when form has been submitted successfully. Please take a look.

**Review by Sanji515 — APPROVED** [review]
Thanks @drepram for the fix. This can be merged now! :tada:

**Comment by drepram** [comment]
Ah okay, I just hope it wouldn't take long for someone to merge this so I can work on another task. Cheers!

**Comment by drepram** [comment]
Oh, and don't forget to approve the task on Google Code-in!

**Review by RishabhJain2018 — APPROVED** [review]

**Comment by RishabhJain2018** [comment]
LGTM :)
Thanks @drepram and @Sanji515 🎉

**Pull request merged**